### ISSUE TITLE
Seemingly non-deterministic reaction generation and degeneracy calculation

### ISSUE THREAD

**Issue by KEHANG** [ReactionMechanismGenerator/RMG-Py#1232]
I found this non-deterministic behavior by accident, was wondering if anyone has an explanation. Basically, if `generateReactionsFromFamilies()` is called twice using same arguments (reactants, products), the `findDegeneracy()` gives different results. Please see the testing script.

```python
from rmgpy import settings
from rmgpy.data.rmg import RMGDatabase
from rmgpy.molecule import Molecule
from rmgpy.rmg.react import findDegeneracies
# load RMG database to create reactions
database = RMGDatabase()

database.load(
    path = settings['database.directory'], 
    thermoLibraries = ['primaryThermoLibrary'], # can add others if necessary
    kineticsFamilies = 'all', 
    reactionLibraries = [], 
    kineticsDepositories = ''
)
thermodb = database.thermo
# Add training reactions
for family in database.kinetics.families.values():
    family.addKineticsRulesFromTrainingSet(thermoDatabase=thermodb)
# average up all the kinetics rules
for family in database.kinetics.families.values():
    family.fillKineticsRulesByAveragingUp()

r1 = Molecule().fromSMILES('c1ccccc1[CH]CCCCC')
r2 = Molecule().fromSMILES('c1ccccc1[CH]CCCCC')
p1 = Molecule().fromSMILES('c1ccccc1C(CCCCC)C(c2ccccc2)CCCCC')
reactants = [r1, r2]
products = [p1]
new_rxns = database.kinetics.generateReactionsFromFamilies(reactants=reactants, 
                                                           products=products, 
                                                           only_families=['R_Recombination'])

new_rxns = findDegeneracies(new_rxns)
print new_rxns[0].degeneracy

new_rxns = database.kinetics.generateReactionsFromFamilies(reactants=reactants, 
                                                           products=products, 
                                                           only_families=['R_Recombination'])

new_rxns = findDegeneracies(new_rxns)
print new_rxns[0].degeneracy
```

**Comment by mliu49** [comment]
Tried to do a little debugging, but haven't been able to figure out the reason yet. Here are some of my observations so far:
1. It seems to be due to `generateReactionsFromFamilies` rather than `findDegeneracies`. The reason I say this is because the first call to `generateReactionsFromFamilies` ultimately gives degeneracy=1, whereas any subsequent calls gives degeneracy=0.5, *regardless of the order in which `findDegeneracies` is called*. 
2. Based on the first point, I tried to see if there was anything different about the two reactions returned from the first call compared to the two reactions returned from the second call. I looked at memory addresses, isomorphism, and atom ids, but nothing stood out as differentiating the results of the first call and the results of the second call.
